File: 프로그래머스/연습문제/Lv1/카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    idx1,idx2,idx_goal = 0,0,0
    cards1.append(" ")
    cards2.append(" ")

    while idx_goal < len(goal):
        if cards1[idx1]!=goal[idx_goal] and cards2[idx2]!=goal[idx_goal]:
            return "No"
        elif cards1[idx1] == goal[idx_goal]: idx1 += 1
        else: idx2 += 1

        idx_goal += 1

    return "Yes"

File: 프로그래머스/연습문제/Lv1/카드 뭉치/카드 뭉치.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> cards1, vector<string> cards2, vector<string> goal) {
    int idx1=0, idx2=0, idx_goal=0;
    cards1.push_back(" ");
    cards2.push_back(" ");
    
    while(idx_goal<goal.size()){
        if(cards1[idx1]!=goal[idx_goal] && cards2[idx2]!=goal[idx_goal])
            return "No";
        else if(cards1[idx1] == goal[idx_goal]) idx1++;
        else idx2++;
        
        idx_goal++;
    }
    
    return "Yes";
}
